internal/handlers/tigris/msg_update.go

```diff
@@ -16,11 +16,202 @@ package tigris
+	}
+	var ok bool
+	if fp.collection, ok = collectionParam.(string); !ok {
+		return nil, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("collection name has invalid type %s", common.AliasFromType(collectionParam)),
+		)
+	}
+
+	var updates *types.Array
+	if updates, err = common.GetOptionalParam(document, "updates", updates); err != nil {
+		return nil, err
+	}
+
+	// created, err := h.pgPool.CreateTableIfNotExist(ctx, fp.db, fp.collection)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// if created {
+	// 	h.L.Info("Created table.", zap.String("schema", fp.db), zap.String("table", fp.collection))
+	// }
+
+	var matched, modified int32
+	var upserted types.Array
+	for i := 0; i < updates.Len(); i++ {
+		update, err := common.AssertType[*types.Document](must.NotFail(updates.Get(i)))
+		if err != nil {
+			return nil, err
+		}
+
+		unimplementedFields := []string{
+			"c",
+			"multi",
+			"collation",
+			"arrayFilters",
+			"hint",
+		}
+		if err := common.Unimplemented(update, unimplementedFields...); err != nil {
+			return nil, err
+		}
+
+		var q, u *types.Document
+		var upsert bool
+		if q, err = common.GetOptionalParam(update, "q", q); err != nil {
+			return nil, err
+		}
+		if u, err = common.GetOptionalParam(update, "u", u); err != nil {
+			return nil, err
+		}
+		if u != nil {
+			if err = common.ValidateUpdateOperators(u); err != nil {
+				return nil, err
+			}
+		}
+
+		if upsert, err = common.GetOptionalParam(update, "upsert", upsert); err != nil {
+			return nil, err
+		}
+
+		fetchedDocs, err := h.fetch(ctx, fp)
+		if err != nil {
+			return nil, err
+		}
+
+		resDocs := make([]*types.Document, 0, 16)
+		for _, doc := range fetchedDocs {
+			matches, err := common.FilterDocument(doc, q)
+			if err != nil {
+				return nil, err
+			}
+
+			if !matches {
+				continue
+			}
+
+			resDocs = append(resDocs, doc)
+		}
+
+		if len(resDocs) == 0 {
+			if !upsert {
+				// nothing to do, continue to the next update operation
+				continue
+			}
+
+			doc := q.DeepCopy()
+			if _, err = common.UpdateDocument(doc, u); err != nil {
+				return nil, err
+			}
+			if !doc.Has("_id") {
+				must.NoError(doc.Set("_id", types.NewObjectID()))
+			}
+
+			must.NoError(upserted.Append(must.NotFail(types.NewDocument(
+				"index", int32(0), // TODO
+				"_id", must.NotFail(doc.Get("_id")),
+			))))
+
+			if err = h.insert(ctx, fp, doc); err != nil {
+				return nil, err
+			}
+
+			matched++
+			continue
+		}
+
+		matched += int32(len(resDocs))
+
+		for _, doc := range resDocs {
+			changed, err := common.UpdateDocument(doc, u)
+			if err != nil {
+				return nil, err
+			}
+
+			if !changed {
+				continue
+			}
+
+			res, err := h.update(ctx, fp, doc)
+			if err != nil {
+				return nil, err
+			}
+			modified += int32(res)
+		}
+	}
+
+	res := must.NotFail(types.NewDocument(
+		"n", matched,
+	))
+	if upserted.Len() != 0 {
+		must.NoError(res.Set("upserted", &upserted))
+	}
+	must.NoError(res.Set("nModified", modified))
```

Comment:
So, in the case of `insert` his value will be 0 (and `matched` will be 1)? I guess it's on purpose, I just want to double-check that this was the idea.

Reply:
It is the same code as in pg handler, and it (probably) has the same bug: #571